### PR TITLE
feat: Support stop serve

### PR DESCRIPTION
**Pull request by nfwyst**
Hi, I added some functionality to check weather the service is running, otherwise manage serve starting by job, thats good when i know i do not need a background process after exiting neovim. and fix typo of mistral, thanks

### CODE REVIEW

File: lua/gen/init.lua [hunk 1]
```diff
@@ -54,12 +54,28 @@ end
 M.command = 'ollama run $model $prompt'
 M.model = 'mistral:instruct'
 
+local function is_serving(pname)
+    local handle, ok = pcall(io.popen, 'ps aux | grep "' .. pname .. '" | grep -v grep')
```
Comment (alaaibrahim) [inline lua/gen/init.lua:58]: Why not use pgrep
```suggestion
    local handle, ok = pcall(io.popen, 'pgrep "' .. pname .. '"')
    if not ok or not handle then
      return false
    end
    handle:close()
    return true
```

This should work on both gnu and bsd systems, and you don't need to read the output

File: lua/gen/init.lua [hunk 2]
```diff
@@ -54,12 +54,28 @@ end
...
+    if not ok or not handle then
+      return false
+    end
+    local result = handle:read("*a")
+    handle:close()
+    return result ~= ""
+end
```
Comment (alaaibrahim) [inline lua/gen/init.lua:65]:
```suggestion
local function is_serving(pname)
    local handle, ok = pcall(io.popen, 'pgrep -q "' .. pname .. '"')
    if not ok or not handle then
      return false
    end
    handle:close()
    return true
end
```

This should be supported on both bsd (including macos) and gnu (including linux), and no need for all the greps

Comment (alaaibrahim) [reply lua/gen/init.lua:65]: Sorry, didn't mean to intervene, just a suggestion.

**Comment by alaaibrahim** [comment]
> Hi @nfwyst ,
> 
> Thank you for this PR, this will only work on unix systems, not on windows, right?

yes, I don't think this would work on windows, although ollama doesn't too. If windows support ever became an issue, this would need to be revisited, and maybe think of a better way to do it other than looking for an ollama process

File: lua/gen/init.lua (hunk 1 above)
Comment (alaaibrahim) [inline lua/gen/init.lua:58]: https://www.lua.org/pil/8.4.html
I believe the return values are reversed for `pcall`, `ok` should be before `handle`

**Comment by nfwyst** [comment]
@David-Kunz I made some code change, should work on windows now, thanks! and thank you bro @alaaibrahim too.

**Review by alaaibrahim — APPROVED** [review]

**Pull request closed without merging**